### ISSUE TITLE
pfnSetModel does not warn about missed precaches in most cases

### ISSUE THREAD

**Issue by x6herbius** [FWGS/xash3d-fwgs#19]
Just quickly noting this down before I forget, as I'm in the middle of other things.

I discovered today that I was forgetting to precache a model in some of my own game code, but only after having stepped through `pfnSetModel` in the debugger. Specifically, the following code seems to be to be incorrect:

``` c
// check to see if model was properly precached
for( i = 1; i < MAX_MODELS && sv.model_precache[i][0]; i++ )
{
    if( !Q_stricmp( sv.model_precache[i], name ))
        break;
}

if( i == MAX_MODELS )
{
    Con_Printf( S_ERROR "no precache: %s\n", name );
        return;
}
```

`i` incremented past the last precached model (which in my case was index 100), so the model returned was null. Instead of hitting the error log and returning, it silently set a blank model on my entity because `i` was still within the entire list bounds.

If there's a count for how many models have been precached in `sv.model_precache`, I think checking against this instead of `MAX_MODELS` would fix the issue.

**Comment by a1batross** [comment]
Thanks. I have reported this bug to Unkle Mike to be in sync and will push fix a bit later.

**Comment by a1batross** [comment]
Fixed in 4d63eb4413ba2ca69cfd018acf25584feb9ac807
